Source/Randomizer.cpp

```diff
@@ -403,11 +403,15 @@ void Randomizer::RandomizeChallenge() {
 void Randomizer::RandomizeAudioLogs() {
     std::vector<int> randomOrder(audiologs.size(), 0);
     std::iota(randomOrder.begin(), randomOrder.end(), 0);
-    Randomize(randomOrder, SWAP::NONE);
+    RandomizeInPlace(randomOrder, SWAP::NONE);
     ReassignNames(audiologs, randomOrder);
 }
 
-void Randomizer::Randomize(std::vector<int>& panels, int flags) {
+void Randomizer::Randomize(std::vector<int> panels, int flags) {
```

Comment:
Okay, I think I understand my confusion. Under what circumstances do we *not* want to make a copy? It seems like we aren't re-using any of these lists after they've been shuffled.

Reply:
There's a pattern used in a few methods (RandomizeMountain, RandomizeJungle, ...) where a vector of increasing numbers is generated and then this method is used to shuffle them around. Then that shuffled vector is used to change the order that certain panels activate in. But I feel like that could be a separate method that doesn't have any of the panel swapping logic in it. (Which is how I did it in the followup change that I haven't quite finished yet: https://github.com/hatkirby/witness-randomizer/commit/d3fbde6fec695215165345a06e9b77716e19b2ee#diff-9392389c78b463a8eeead50a46276193a720b463bf44543eb8d620431dcc1ccfR412)

Reply:
That makes sense to me. I would just replace all of the calls to `Randomize(*, Swap::NONE)` with `Shuffle(*)` -- and then have Shuffle pass-by-reference, and make Randomize pass-by-value.
```cpp
// Range is [start, end)
void Randomizer::Shuffle(std::vector<int> &order, size_t startIndex, size_t endIndex) {
    if (panels.size() == 0) return;
    if (startIndex >= endIndex) return;
    if (endIndex >= order.length()) endIndex = static_cast<int>(order.length());
    for (size_t i = endIndex-1; i > startIndex; i--) {
        const int target = Random::RandInt(static_cast<int>(startIndex), static_cast<int>(i));
        std::swap(order[i], order[target]);
    }
}
```